src/devices/Card/Ndef/NdefMessage.cs

```diff
@@ -18,6 +18,70 @@ public class NdefMessage
         /// </summary>
         public const byte GeneralPurposeByteNdefVersion = 0b0100_0000;
 
+        /// <summary>
+        /// From a raw message, find the start and stop of an NDEF message
+        /// </summary>
+        /// <param name="toExtract">The byte array where the message is</param>
+        /// <returns>The start and end position</returns>
+        public static (int Start, int Size) GetStartSizeNdef(Span<byte> toExtract)
+        {
+            int idx = 0;
+            // Check if we have 0x03 so it's a possible, NDEF Entry
+            while (idx < toExtract.Length)
+            {
+                if (toExtract[idx++] == 0x03)
+                {
+                    break;
+                }
+            }
```

Comment:
I'm confused by this. In any buffer with random bytes there is a chance there will be 3 somewhere in there. Is there any reliable way to know at which position the 3 should be?

Reply:
The check is done later with the size of the buffer, and the terminator. I do agree that it's not perfect, that said, in theory, the buffer should start with 0x03 or being part of the very first one. The NDEF specifications do not gives a lot of flexibility for that. So it has to be there or it's not an NDEF buffer (0x03 then length and after length 0xFE)